src/antelope/stores/nfts.ts

```diff
@@ -5,241 +5,211 @@
-            // replace all of this logic with real fetch & set
-            const nfts: ShapedNFT[] = Object.keys(this.__nfts[label].contracts).map(
-                (contractAddress: string) => this.__nfts[label].contracts[contractAddress],
-            ).flat();
-            this.setAccountNfts(label, address, nfts);
-        },
-        async getNftDetails(label: Label, contract: string, id: string) {
-            // initial state temporarily contains all mock data
-            // need to handle nft not found, perhaps log antelope error
-            // await this.getNftsFromContract(label, contract);
+        async updateNFTsForAccount(label: string, account: AccountModel | null) {
+            this.trace('updateNFTsForAccount', label, account);
+            if (!account?.account) {
+                return;
+            }
+
+            const owner = account.account;
+
+            // we initialize the inventory for this label or take the existing one
+            this.__inventory[label] = this.__inventory[label] || {
+                list: [],
+                loading: true,
+            };
 
-            const nftsForContract = this.__nfts[label].contracts[contract] || [];
+            try {
+                useFeedbackStore().setLoading('updateNFTsForAccount');
+                const chain = useChainStore().getChain(label);
+                if (chain.settings.isNative()) {
+                    const chain_settings = chain.settings as NativeChainSettings;
+                    chain_settings.getNFTsInventory(owner, toRaw(this.__filter)).then((nfts) => {
+                        this.__inventory[label].list = nfts;
+                        this.__inventory[label].loading = false;
+                        this.trace('updateNFTsForAccount', 'indexer returned:', nfts);
+                        useFeedbackStore().unsetLoading('updateNFTsForAccount');
+                    });
+                } else {
+                    const chain_settings = chain.settings as EVMChainSettings;
 
-            return nftsForContract.find(({ id: nftId }) => nftId === id);
+                    if (chain_settings.hasIndexerSupport()) {
+                        chain_settings.getNFTsInventory(owner, toRaw(this.__filter)).then((nfts) => {
+                            this.__inventory[label].list = nfts;
+                            this.__inventory[label].loading = false;
+                            this.trace('updateNFTsForAccount', 'indexer returned:', nfts);
+                            useFeedbackStore().unsetLoading('updateNFTsForAccount');
+                            nfts.forEach((nft) => {
+                                nft.watch(() => {
+                                    // this forces an update for watchers to be called
+                                    this.__inventory[label].list = [...nfts];
+                                });
+                            });
+                        });
```

Comment:
This workaround was necessary to make the inventory list update when the NFT internal data changes. I don't know why is not happening directly so I did this to make it work.